packages/universal-components/src/RangeDatePicker/libs.js

```diff
@@ -49,12 +66,217 @@ export const getMonthMatrix = (
+  fingerRelativePosition: { x: number, y: number },
+  months: Array<AnotatedMonthType>,
+  monthIndex: number,
+  weekIndex: number,
+  dayItemSize: DayItemSizeType,
+) => {
+  const touchBuffer = getTouchBuffer(dayItemSize.height);
+  const isDraggingUp = fingerRelativePosition.y < 0;
+  const properSign = isDraggingUp ? -1 : 1;
+  const fingerYPositionWithoutCurrentMonth = isDraggingUp
+    ? Math.abs(fingerRelativePosition.y) - weekIndex * dayItemSize.height
+    : Math.abs(fingerRelativePosition.y) -
+      (months[monthIndex].weeks.length - weekIndex - 1) * dayItemSize.height;
+
+  const isShiftOnlyInsideOfCurrentMonth =
+    fingerYPositionWithoutCurrentMonth < spaceBetweenMonths + touchBuffer;
+
+  if (isShiftOnlyInsideOfCurrentMonth) {
+    return 0;
+  }
+  // @TODO Optimise case when user is dragging between borders of the particular months. It should mark only the first/last day of the month.
+
+  const monthSpacesOrder = isDraggingUp
+    ? months.slice(0, monthIndex).reverse()
+    : months.slice(monthIndex);
+
+  const monthSpacesInfo = monthSpacesOrder.reduce(
+    (acc, month) => {
+      if (
+        month.weeks.length * dayItemSize.height +
+          (spaceBetweenMonths + touchBuffer) <
+        acc.remains
+      ) {
+        return {
+          remains: acc.remains - (spaceBetweenMonths + touchBuffer),
+          numberOfMonths: acc.numberOfMonths + 1,
+        };
+      }
+      return acc;
+    },
+    {
+      remains: fingerYPositionWithoutCurrentMonth,
+      numberOfMonths: 1,
+    },
+  );
+
+  return properSign * monthSpacesInfo.numberOfMonths;
+};
+
+export const getTouchBuffer = (dayItemHeight: number) =>
+  dayItemHeight * touchBufferRatio;
+
+export const setProperDateInterval = (
+  grabbedSide: GrabbedSideType,
+  newDateWithXYShift: Date,
+  grabbedDay: Date,
+  selectedDates: $ReadOnlyArray<Date>,
+) => {
+  if (grabbedSide === 'left') {
+    if (isBefore(newDateWithXYShift, grabbedDay)) {
+      return [newDateWithXYShift, selectedDates[1]];
+    } else if (isBefore(newDateWithXYShift, selectedDates[1])) {
+      return [newDateWithXYShift, selectedDates[1]];
+    } else if (isSameDay(newDateWithXYShift, selectedDates[1])) {
+      return [selectedDates[1], selectedDates[1]];
+    }
+    return [selectedDates[1], newDateWithXYShift];
+  } else if (isAfter(newDateWithXYShift, grabbedDay)) {
+    return [selectedDates[0], newDateWithXYShift];
+  } else if (isAfter(newDateWithXYShift, selectedDates[0])) {
+    return [selectedDates[0], newDateWithXYShift];
+  } else if (isSameDay(newDateWithXYShift, selectedDates[0])) {
+    return [selectedDates[0], selectedDates[0]];
+  }
+  return [newDateWithXYShift, selectedDates[0]];
+};
+
+// @TODO add flow type for event
+export const findRelatedItem = (
+  event: any,
+  grabbedDay: Date,
+  selectedDates: $ReadOnlyArray<Date>,
+  dayItemSize: DayItemSizeType,
+  grabbedSide: GrabbedSideType,
+  weekStartsOn: WeekStartsType,
+  callback: (Array<Date>) => void,
```

Comment:
this method has too many parameters, it will be hard to use in the future. Could it take an object instead? 

You could consider doing it with other functions in this file as well.

Reply:
Good point, done